### PR TITLE
Tests for S4200: Add testcases from S4214

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Part of #6488

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp11.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Runtime.InteropServices;
+
+public interface IInterface
+{
+    [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+    static extern virtual bool RemoveDirectory1(string name); // Noncompliant
+
+    [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+    public static extern virtual bool RemoveDirectory2(string name);  // Noncompliant
+}
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp11.cs:10]: Copy of https://github.com/SonarSource/sonar-dotnet/blob/c18ed76dbbc997007e41be9e9e49df3aa68ab233/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PInvokesShouldNotBeVisible.CSharp11.cs#L1-L10

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp9.cs [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
+        public void Method()
+        {
+            [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+            static extern bool RemoveDirectory(string name);     // Compliant - Method is not publicly exposed
+        }
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        public static extern bool RemoveDirectory(string name);  // Noncompliant
+    }
+}
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp9.cs:31]: Copy of https://github.com/SonarSource/sonar-dotnet/blob/c18ed76dbbc997007e41be9e9e49df3aa68ab233/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PInvokesShouldNotBeVisible.CSharp9.cs#L1-L13

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.cs [hunk 3]
```diff
@@ -88,3 +88,37 @@ internal class PrivateClass2
...
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)] // Error [CS0601]
+        public static bool RemoveDirectory7(string name); // Error [CS0501] - so do not raise an issue
+    }
+
+    internal class Foo
+    {
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        public static extern bool RemoveDirectory1(string name); // Compliant because effective accessibility is not public
+    }
+}
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.cs:124]: Copy of https://github.com/SonarSource/sonar-dotnet/blob/c18ed76dbbc997007e41be9e9e49df3aa68ab233/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PInvokesShouldNotBeVisible.cs#L1-L36

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.cs [hunk 4]
```diff
@@ -88,3 +88,37 @@ internal class PrivateClass2
...
+        internal protected static extern bool RemoveDirectory3(string name); // Noncompliant
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        internal static extern bool RemoveDirectory4(string name);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)]
+        private static extern bool RemoveDirectory5(string name);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode)] // Error [CS0601]
+        public extern bool RemoveDirectory6(string name);      // Noncompliant
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.cs:113]: Was compliant before: `// Error [CS0601] - so do not raise an issue`

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**